tests/tests_unit/test_base.py

```diff
@@ -204,6 +211,61 @@ def test_use_method_which_requires_cognite_client__client_not_set(self):
         with pytest.raises(CogniteMissingClientError):
             mr.use()
 
+    @pytest.mark.dsl
+    @pytest.mark.parametrize(
+        "cognite_resource_subclass",
+        [
+            pytest.param(class_, id=f"{class_.__name__} in {class_.__module__}")
+            for class_ in all_concrete_subclasses(CogniteResource)
+        ],
+    )
+    def test_json_serialize(self, cognite_resource_subclass: type[CogniteResource], cognite_mock_client):
+        instance = FakeCogniteResourceGenerator(seed=42, cognite_client=cognite_mock_client).create_instance(
+            cognite_resource_subclass
+        )
+
+        dumped = instance.dump(camel_case=True)
+        json_serialised = json.dumps(dumped)
+        json_deserialised = json.loads(json_serialised)
+        loaded = instance.load(json_deserialised, cognite_client=cognite_mock_client)
```

Comment:
Regarding the "expensive to instantiate" comment, as long as we don't need all the capabilities of the fully specced cog-client-mock, something simple as this might do the trick:
```py
@pytest.fixture
def mock_client():
    # We allow the mock to pass isinstance checks
    (client := MagicMock()).__class__ = CogniteClient
    return client
```